LGIA geoid grid

yurish opened the pull request:
Adding Latvian Geospatial Information Agency grid for conversion of LKS-92 (EPSG:4669) to Latvia 2000 height (EPSG:7700)

jjimenezshaw commented:
Opening the tif file in QGIS I see two problems:
 - 9999 is not set as no-data value. (just opening with QGIS)
 - It is rotated and at a wrong location. Apparently you swapped latitude with longitude

How did you convert the data? It is a good idea to have a script to easily reproduce it. This is one example: https://github.com/OSGeo/PROJ-data/blob/master/fi_nls/build_FIN2000_05_23.sh

Also some test with some checkpoints (using https://github.com/OSGeo/PROJ/pull/4412/)

jjimenezshaw commented:
Running the script to update index.html you are using a version that is not master. You are deleting some info related to EPSG:10689 (Finland)
It is better if you use the last docker image. It needs git, so some hacks are needed. I think this may work:
```
docker pull ghcr.io/osgeo/gdal:ubuntu-full-latest
docker run --workdir $PWD --rm -v /home:/home ghcr.io/osgeo/gdal:ubuntu-full-latest sh -c "apt update && apt install -y git && git config --global --add safe.directory $PWD && python3 regenerate_index_html.py"
```

yurish commented:
Thank you for your comments, @jjimenezshaw ! I have fixed the grid and added the script, that generates it. Can you guide me, what is the best solution for adding tests for the grid?

jjimenezshaw commented on lv_lgia/lv_14_convert.sh:
```diff
@@ -0,0 +1,44 @@
+gdal_translate -of GTiff -a_nodata 9999 "$TEMP_FILE" "$OUTPUT_TIF"
+
+# Run the vertoffset grid conversion
+../grid_tools/vertoffset_grid_to_gtiff.py \
```
add `python3` before, as you do in the next command

jjimenezshaw commented on lv_lgia/lv_14_convert.sh:
```diff
@@ -0,0 +1,44 @@
+  --parameter-name "geoid_undulation" \
+  --source-crs "EPSG:4949" \
+  --target-crs "EPSG:7700" \
+  "$OUTPUT_TIF" "$OUTPUT_TIF"
```
Using the same name for input and output is not a good idea. Just use another temp file.

jjimenezshaw commented on files.geojson:
```diff
@@ -133,9 +133,9 @@
 { "type": "Feature", "properties": { "url": "https://cdn.proj.org/eur_nkg_nkgrf03vel_realigned.tif", "name": "eur_nkg_nkgrf03vel_realigned.tif", "area_of_use": "Nordic and Baltic countries", "type": "VELOCITY", "source_crs_name": "Unknown based on GRS80", "source": "The Nordic Geodetic Commission", "source_country": "Europe", "source_id": "eur_nkg", "source_url": "http://www.nordicgeodeticcommiss ...
 { "type": "Feature", "properties": { "url": "https://cdn.proj.org/eur_nkg_nkgrf17vel.tif", "name": "eur_nkg_nkgrf17vel.tif", "area_of_use": "Nordic and Baltic countries", "type": "VELOCITY", "source_crs_code": "EPSG:8403", "source_crs_name": "ETRF2014", "source": "The Nordic Geodetic Commission", "source_country": "Europe", "source_id": "eur_nkg", "source_url": "http://www.nordicgeodeticcommissio ...
 { "type": "Feature", "properties": { "url": "https://cdn.proj.org/fi_nls_README.txt", "name": "fi_nls_README.txt", "source": "National Land Survey of Finland (MML)", "source_country": "Finland", "source_id": "fi_nls", "source_url": "https://www.maanmittauslaitos.fi", "file_size": 3458, "sha256sum": "05708b917dd04f8704911c48c6aa0b3eb26643eaa9403bea7157c3563f0223e5" }, "geometry": null },
-{ "type": "Feature", "properties": { "url": "https://cdn.proj.org/fi_nls_fin2000.tif", "name": "fi_nls_fin2000.tif", "area_of_use": "Finland", "type": "VERTICAL_OFFSET_GEOGRAPHIC_TO_VERTICAL", "source_crs_code": "EPSG:10689", "source_crs_name": "EUREF-FIN", "target_crs_code": "EPSG:5717", "target_crs_name": "N60 height", "source": "National Land Survey of Finland (MML)", "source_country": "Finlan ...
```
Please, use the proper version of PROJ. The gdal docker image is a good option.

jjimenezshaw commented:
> Can you guide me, what is the best solution for adding tests for the grid?

In the PROJ PR, add a couple of checkpoints and the command line to convert them to see they are correct. It can be in the PR description itself, nothing fancy

rouault commented:
I've done the adjustments suggested by @jjimenezshaw  and pushed them in the branch. The grid seems to be consistent with EGM2008:

$ echo 24 57 0 | bin/cct +proj=vgridshift +grids=/home/even/proj/PROJ-data/lv_lgia/lv_lgia_lv14.tif
 24.0000000000   57.0000000000      -20.7065           inf

$ echo 24 57 0 | bin/cct +proj=vgridshift +grids=/home/even/proj/PROJ-data/us_nga/us_nga_egm08_25.tif
 24.0000000000   57.0000000000      -20.9804           inf

$ echo 23 57 0 | bin/cct +proj=vgridshift +grids=/home/even/proj/PROJ-data/lv_lgia/lv_lgia_lv14.tif
 23.0000000000   57.0000000000      -21.6148           inf

$ echo 23 57 0 | bin/cct +proj=vgridshift +grids=/home/even/proj/PROJ-data/us_nga/us_nga_egm08_25.tif
 23.0000000000   57.0000000000      -21.8943           inf

yurish commented:
Thank you, @rouault ! Same coordinates from the official LGIA calculator (https://www.lgia.gov.lv/en/koordinatu-parrekinu-kalkulators):

57 24 0 -20.706
57 23 0 -21.614
